### PR TITLE
[WIP] refactor Ruby Event getter and setter

### PR DESCRIPTION
**Pull request by colinsurprenant**
This is a WIP for the Event  getter and setter refactor. 

Per #5141 and #5140 

This code is pushed in `feature/event_interface` **shared branch**.

For this to work the plugins using a local `:path` in the included modified `Gemfile` need to be tweeked to use the new interface. 

- [x] Modify the Ruby Event interface
- [x] Pass core specs using Ruby Event
- [x] Modify Java Event interface
- [x] Pass core specs using Java Event
- [x] bump logstash-core-plugin-api for new Event interface
- [x] publish modified plugins required for core specs using the new logstash-core-plugin-api version

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -3,23 +3,27 @@
...
 gem "ci_reporter_rspec", "1.0.0", :group => :development
 gem "simplecov", :group => :development
 gem "coveralls", :group => :development
-# Tins 1.7 requires the ruby 2.0 platform to install,
-# this gem is a dependency of term-ansi-color which is a dependency of coveralls.
-# 1.6 is the last supported version on jruby.
 gem "tins", "1.6", :group => :development
 gem "rspec", "~> 3.1.0", :group => :development
-gem "logstash-devutils", "~> 0.0.15", :group => :development
+gem "logstash-devutils", ">= 0", :path => "/Users/colin/dev/src/elasticsearch/logstash-devutils"
```
Comment (suyograo) [inline Gemfile]: User specific path? :)

Comment (colinsurprenant) [inline Gemfile]: yes, this is not a mergeable WIP and I added in the description that these must be addressed...

**Comment by colinsurprenant** [comment]
@ph @suyograo we need to discuss the strategy for bumping logstash-core-plugin-api + branching so we can start updating the plugins ...

**Comment by ph** [comment]
@colinsurprenant I think the following strategy will work:
- create a logstash-core-plugin-api 2.0.0
- make a branch on every plugin and update them to use this gem.

I guess we need to bump major version of the plugin two?

**Comment by suyograo** [comment]
+1 to what @ph said. Also update the gemspec for all plugins

**Comment by ph** [comment]
@suyograo @colinsurprenant we should also be able to release the new plugins with their major version and test them. People wont be able to install them in version prior to < 5.0.

**Comment by colinsurprenant** [comment]
+1

**Comment by colinsurprenant** [comment]
jenkins please test this

**Comment by colinsurprenant** [comment]
jenkins, please test this

**Comment by colinsurprenant** [comment]
jenkins you mothafucka test this shit now

**Comment by ph** [comment]
Test are passing, all the update plugins are also passing.

LGTM.

**Comment by colinsurprenant** [comment]
ok, so will push commit to remove branches plugins gem, then we can merge in master

**Comment by colinsurprenant** [comment]
rebasing and merging.

**Comment by colinsurprenant** [comment]
merged into master and 5.0

**Pull request merged**